pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -571,6 +580,15 @@ func (cr *PerconaXtraDBCluster) CheckNSetDefaults(serverVersion *version.ServerV
 		}
 	}
 
+	if c.PXC.Expose.Enabled {
+		if c.PXC.Expose.TrafficPolicy == "" {
+			c.PXC.Expose.TrafficPolicy = corev1.ServiceExternalTrafficPolicyTypeLocal
+		}
+		if c.PXC.Expose.Type == "" {
+			c.PXC.Expose.Type = corev1.ServiceTypeClusterIP
+		}
```

Comment:
it is better to borrow logic from PSMDB Operator service-per-pod
https://github.com/percona/percona-server-mongodb-operator/blob/main/pkg/psmdb/service.go#L88-L97